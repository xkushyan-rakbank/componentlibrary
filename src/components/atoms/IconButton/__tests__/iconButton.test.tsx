import '@testing-library/jest-dom'

import { IconButtonProps as MuiIconButtonProps } from '@mui/material'
import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
//import primaryPlusIcon from '../../../../assets/primaryPlusIcon.svg'
import IconButtons from '../IconButton'
import PersonIcon from '@mui/icons-material/Person'
describe('IconButtons', () => {
  const defaultProps: MuiIconButtonProps = {
    //icon: <span>Test Icon</span>,
    children: <PersonIcon />,
    onClick: jest.fn(),
  }

  const renderComponent = (props: MuiIconButtonProps = defaultProps) => {
    return render(<IconButtons icon={props.children} {...props} />)
  }

  test.only('Should match snapshot', () => {
    expect(renderComponent().container).toMatchSnapshot()
  })

  test('Should renders Icon Button component with children', () => {
    renderComponent()
    expect(screen.getByTestId('PersonIcon')).toBeTruthy()
  })

  test('Should call onclick fuction on icon button click', () => {
    renderComponent()
    const btn = screen.getByRole('button')

    screen.debug(btn)
    fireEvent.click(btn)
    expect(defaultProps.onClick).toHaveBeenCalled()
  })
  test('Should disabled button when disabled prop is true', () => {
    const props = {
      ...defaultProps,
      disabled: true,
    }
    renderComponent(props)
    const btn = screen.getByRole('button')

    expect(btn).toHaveProperty('disabled', true)
  })
})
