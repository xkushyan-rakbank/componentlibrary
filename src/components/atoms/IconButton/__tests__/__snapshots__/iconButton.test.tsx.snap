// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtons Should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    icon="[object Object]"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4m0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
